### PR TITLE
Fix decoder with attention example in the README

### PR DESCRIPTION
**Pull request by guillaumekln**
There were several issues:

* missing encoder masking
* missing training flag
* `encoder_state` passed as memory instead of `encoder_outputs`

This commit also improves variables name consistency and include attention in the first example to have one complete training usage.

Related to #1995.

### CODE REVIEW

File: tensorflow_addons/seq2seq/README.md [hunk 1]
```diff
@@ -60,26 +71,38 @@ import tensorflow_addons as tfa
...
 
 # Sampler
 sampler = tfa.seq2seq.sampler.TrainingSampler()
 
 # Decoder
-decoder_cell = tf.keras.layers.LSTMCell(num_units)
 projection_layer = tf.keras.layers.Dense(num_outputs)
-decoder = tfa.seq2seq.BasicDecoder(
-    decoder_cell, sampler, output_layer=projection_layer)
+decoder = tfa.seq2seq.BasicDecoder(decoder_cell, sampler, output_layer=projection_layer)
```
Comment (qlzh727) [inline tensorflow_addons/seq2seq/README.md:96]: Seems to be a long line, can we wrap it?

Comment (guillaumekln) [reply tensorflow_addons/seq2seq/README.md:96]: I actually reformatted the example using `black` to be consistent with the codebase itself.
